### PR TITLE
Implement `wal-e wal-fetch` compatible bash script

### PR DESCRIPTION
**Pull request by CyberDem0n**
and use it by default for AWS deployments

### CODE REVIEW

File: postgres-appliance/wal-e-wal-fetch.sh [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+else
+    readonly SIGNED_HEADERS="host;x-amz-content-sha256;x-amz-date;x-amz-security-token"
+    readonly REQUEST_TOKEN="x-amz-security-token:$AWS_SECURITY_TOKEN\n"
+    readonly TOKEN_HEADER=(-H "x-amz-security-token: $AWS_SECURITY_TOKEN")
+fi
+
+function s3_get() {
+    local segment=$1
+    local destination=$2
+    local FILE=$BUCKET_PATH/wal_005/$segment.lzo
```
Comment (a1exsh) [inline postgres-appliance/wal-e-wal-fetch.sh]: Is it always `wal_005`?

Comment (CyberDem0n) [inline postgres-appliance/wal-e-wal-fetch.sh]: Yes: https://github.com/wal-e/wal-e/blob/master/wal_e/storage/base.py#L17

File: postgres-appliance/wal-e-wal-fetch.sh [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        --s3-prefix )
+            WALE_S3_PREFIX=$2
+            shift
+            ;;
+        -k|--aws-access-key-id )
+            AWS_ACCESS_KEY_ID=$2
+            shift
+            ;;
+        --aws-instance-profile )
+            AWS_INSTANCE_PROFILE=1
```
Comment (avaczi) [inline postgres-appliance/wal-e-wal-fetch.sh]: isn't a shift missing here?

Comment (CyberDem0n) [inline postgres-appliance/wal-e-wal-fetch.sh]: no

File: postgres-appliance/wal-e-wal-fetch.sh [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    if [[ ! -z $WALE_S3_ENDPOINT && $WALE_S3_ENDPOINT =~ ^([a-z\+]{2,10}://)?(s3-([^\.]+)[^:\/?]+) ]]; then
+        S3_HOST=${BASH_REMATCH[2]}
+        AWS_REGION=${BASH_REMATCH[3]}
+    elif [[ $AWS_INSTANCE_PROFILE == 1 ]]; then
+        load_region_from_aws_instance_profile
+    fi
+fi
+
+if [[ -z $AWS_REGION ]]; then
+    echo AWS_REGION is unknown
```
Comment (alexeyklyukin) [inline postgres-appliance/wal-e-wal-fetch.sh]: exit 1

**Comment by alexeyklyukin** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**